src/common/matmul.cpp

```diff
@@ -312,6 +312,10 @@ status_t matmul_desc_init(matmul_desc_t *matmul_desc,
     VCHECK_MATMUL(
             IMPLICATION(bias_desc, !reduce_desc), VERBOSE_UNSUPPORTED_BIAS_CFG);
 
+    VCHECK_MATMUL(!any_memory_desc_host_scalar(src_desc, weights_desc,
+                          bias_desc, dst_desc, reduce_desc),
+            VERBOSE_UNSUPPORTED_FORMAT_KIND);
```

Comment:
Is updating CPU implementation (or some common spot) is needed not to let host scalar in there?

Reply:
This check is to disallow host-side scalars to be input arguments, as it is not required and if allowed, then it requires special handling for GPU. Is this what you were asking?

Reply:
Yes, host-side scalars memory obj are OK/supported for any CPU implementation (SYCL CPU is also OK), there is no check for that and it is being tested. The special handling (and therefore check) is only required for GPU.

Reply:
OK to resolve?